internal/pkg/cli/init.go

```diff
@@ -540,6 +555,50 @@ func (o *initOpts) askShouldDeploy() error {
 	return nil
 }
 
+func (o *initOpts) askEnvName() error {
+	if o.initVars.envName != "" {
+		return nil
+	}
+	envs, err := o.store.ListEnvironments(*o.appName)
+	if err != nil {
+		return fmt.Errorf("list environments: %w", err)
+	}
+	// Configure prompt options based on number of environments.
+	promptOpts := []prompt.PromptConfig{prompt.WithFinalMessage("Environment name:")}
+	if len(envs) == 0 {
+		promptOpts = append(promptOpts, prompt.WithDefaultInput(defaultEnvironmentName))
+	} else {
+		log.Infoln("It looks like you have existing environments in this application.")
+		// Set up selection options.
+		options := make([]string, len(envs))
+		for e := range envs {
+			options[e] = envs[e].Name
+		}
+		options = append(options, envPromptCreateNew)
+		// Select one of existing envs or create a new one.
+		v, err := o.prompt.SelectOne(initExistingEnvSelectPrompt, initExistingEnvSelectHelp, options)
+		if err != nil {
+			return fmt.Errorf("select environment: %w", err)
+		}
+		// Customer has selected an existing environment. Return early.
```

Comment:
in this case, can't we skip calling `envInit.Execute()` entirely?

Reply:
yeah, and I've reworked the logic to ensure that.